src/machine_learning/loss_function/negative_log_likelihood.rs

```diff
@@ -0,0 +1,128 @@
+// Negative Log Likelihood Loss Function
+//
+// The `neg_log_likelihood` function calculates the Negative Log Likelyhood loss,
+// which is a loss function used for classification problems in machine learning.
+//
+// ## Formula
+//
+// For a pair of actual and predicted values, represented as vectors `y_true` and
+// `y_pred`, the Negative Log Likelihood loss is calculated as:
+//
+// - loss = `-y_true * log(y_pred) - (1 - y_true) * log(1 - y_pred)`.
+//
+// It returns the average loss by dividing the `total_loss` by total no. of
+// elements.
+//
+// https://towardsdatascience.com/cross-entropy-negative-log-likelihood-and-all-that-jazz-47a95bd2e81
+
+pub fn neg_log_likelihood(
+    y_true: &[f64],
+    y_pred: &[f64],
+) -> Result<f64, NegativeLogLikelihoodLossError> {
+    // Checks if the inputs are empty
+    if y_pred.is_empty() || y_true.is_empty() {
+        return Err(NegativeLogLikelihoodLossError::EmptyInputs);
+    }
+    // Checks if the length of the actual and predicted values are equal
+    if y_true.len() != y_pred.len() {
+        return Err(NegativeLogLikelihoodLossError::InputsHaveDifferentLength);
+    }
```

Comment:
```suggestion
    if y_true.len() != y_pred.len() {
        return Err(NegativeLogLikelihoodLossError::InputsHaveDifferentLength);
    }

    if y_pred.is_empty() {
        return Err(NegativeLogLikelihoodLossError::EmptyInputs);
    }
```
Note that at the second `if`, it is sure that the inputs have same lengths. Thanks to this change, the code has lower cyclomatic complexity.